go/vt/vtgate/planbuilder/plancontext/planning_context.go

```diff
@@ -35,6 +36,12 @@ type PlanningContext struct {
 	SemTable     *semantics.SemTable
 	VSchema      VSchema
 
+	// skipBlockJoinArgument tracks BlockBuild operator that should be skipped when
+	// rewriting the operator tree to an AST tree.
+	// This happens when a BlockJoin is pushed under a route, and we do not
+	// need to have a BlockBuild operator anymore on its RHS.
+	skipBlockJoinArgument map[string]any
+
```

Comment:
I believe we need the same as predicate tracker here.

Reply:
We could probably do it in a separate PR, but I rather not do it in this PR. The predicate tracker can change predicates today, but not remove them altogether.